base/abstractarray.jl

```diff
@@ -1130,7 +1130,8 @@ function copymutable(a::AbstractArray)
 end
 copymutable(itr) = collect(itr)
 
-zero(x::AbstractArray{T}) where {T} = fill!(similar(x, typeof(zero(T))), zero(T))
+zero(x::AbstractArray{T}) where {T<:DataType} = fill!(similar(x, typeof(zero(T))), zero(T))
```

Comment:
I think you mean the case where `T isa DataType` here (which cannot be used in a `where` clause though, but would require an `if`/`else` in the same method). And you'd probably want `isconcretetype` instead.

Reply:
Ahh, thank you! That's much better.